app/controllers/search_controller.rb

```diff
@@ -122,21 +122,27 @@ def usernames
   # TODO: [@rhymes] the homepage feed uses `feed_content_search` as an index,
   # we should eventually move it to a JSON result
   # in ArticlesController#Homepage or HomepageController#show
-  # rubocop:disable Metric/PerceivedComplexity
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
   def feed_content
     class_name = feed_params[:class_name].to_s.inquiry
 
     enable_search_2_homepage = (
       class_name.Article? &&
       feed_params[:search_fields].blank? &&
+      feed_params[:sort_by].present? &&
       FeatureFlag.enabled?(:search_2_homepage, current_user)
     )
 
     result =
       if class_name.blank?
-        # If we are in the main feed and not filtering by type return
-        # all articles, podcast episodes, and users
-        feed_content_search.concat(user_search)
+        if FeatureFlag.enabled?(:search_2_articles, current_user)
+          search_postgres_article
+        else
+          # If we are in the main feed and not filtering by type return
+          # all articles, podcast episodes, and users
+          feed_content_search.concat(user_search)
+        end
```

Comment:
The current behavior, with ES, is to join articles and users in the results. We don't do that with PG